tests/test_pyav.py

```diff
@@ -390,3 +398,47 @@ def test_uri_reading(test_images):
     actual = iio.imread(uri, plugin="pyav", index=250)
 
     np.allclose(actual, expected)
+
+
+def test_procedual_writing(test_images):
+    buffer = io.BytesIO()
+    with iio.imopen(buffer, "w", plugin="pyav", extension=".mp4") as file:
+        file.init_video_stream("h264")
+        for frame in iio.imiter(test_images / "cockatoo.mp4", plugin="pyav"):
+            file.write_frame(frame)
+
+    buffer.seek(0)
+    actual = iio.imread(buffer, plugin="pyav", extension=".mp4")
+
+    assert actual.shape == (280, 720, 1280, 3)
+
+
+def test_procedual_writing_with_filter(test_images):
+    buffer = io.BytesIO()
+    with iio.imopen(buffer, "w", plugin="pyav", extension=".mp4") as file:
+        file.init_video_stream("h264", fps=30)
+        file.set_video_filter(
+            filter_sequence=[
+                ("scale", {"size": "vga", "flags": "lanczos"}),
+            ]
+        )
+        for frame in iio.imiter(
+            test_images / "cockatoo.mp4", plugin="pyav", format="yuv444p"
+        ):
+            file.write_frame(frame, pixel_format="yuv444p")
```

Comment:
Are you able to change the pixel format on the fly like this?

Reply:
Yes, though this refers to the incoming pixel format. Maybe we should keep the name `in_pixel_format`? 

The underlying stream for writing has to have the same pixel format throughout and is set via `file.init_video_stream(codec, pixel_format=...)`. All incoming frames are converted to that format before they are encoded.

Reply:
got it

Reply:
If you want to write a file with multiple pixel formats you could create multiple streams and write to each in the respective format. If you set the PTS appropriately you could even write a viewer that can then read that data back. However, this is _very_ non-standard because I don't know of any viewer that recognizes more than one stream.

Also, this is currently not possible with ImageIO, since it is currently designed for exactly one video stream.